src/emailpro/external-contact/add/emailpro-external-contact-add.html

```diff
@@ -1,66 +1,99 @@
+
+                <div class="form-group" data-ng-class="{'has-error': ctrl.externalContactAddForm.emailAddress.$dirty && ctrl.externalContactAddForm.emailAddress.$invalid}">
+                    <label for="emailAddress"
+                           class="control-label col-md-3 required"
                            data-i18n-static="exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_step1_email_label"></label>
                     <div class="col-md-9">
-                        <input class="form-control" id="emailAddress" type="text"
-                               data-ng-model="model.newAccount.externalEmailAddress">
-                        <span class="help-block text-danger"
-                              data-ng-if="!isEmailValid() && model.newAccount.externalEmailAddress"
-                              data-i18n-static="exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_step1_email_invalid"></span>
+                        <input class="form-control"
+                               id="emailAddress"
+                               name="emailAddress"
+                               type="text"
+                               min="0" data-ng-min="0"
+                               max="255" data-ng-max="255"
+                               required
+                               aria-describedby="emailError"
+                               data-ng-pattern="ctrl.emailValidationRegex"
+                               data-ng-model="ctrl.model.externalEmailAddress">
+                            <p id="emailError"
+                               class="help-block"
+                               data-ng-bind-html="ctrl.getEmailErrorMessage()"></p>
                     </div>
                 </div>
-                <div class="form-group" data-ng-if="availableMainDomains">
-                    <label for="availableMainDomain" class="control-label col-md-3 required"
+                <div class="form-group" data-ng-if="ctrl.availableMainDomains != null">
+                    <label for="availableMainDomain"
+                           class="control-label col-md-3 required"
                            data-i18n-static="exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_step1_organization2010_label"></label>
                     <div class="col-md-9">
-                        <select class="form-control input-group-addon" id="availableMainDomain" required
-                                data-ng-options="domain.displayName for domain in availableMainDomains"
-                                data-ng-model="model.attachOrganization2010">
+                        <select class="form-control input-group-addon"
+                                id="availableMainDomain"
+                                name="availableMainDomain"
+                                required
+                                data-ng-options="domain.displayName for domain in ctrl.availableMainDomains"
```

Comment:
You should use `track by` here to improve performances :)

Reply:
Indeed !